nni/nas/oneshot/pytorch/base_lightning.py

```diff
@@ -77,8 +77,12 @@ def traverse_and_mutate_submodules(
     memo = {}
 
     module_list = []
-
+    import types
     def apply(m):
+        # Binding instance methods for nn.Module
+        if not hasattr(m, "sub_state_dict"):
```

Comment:
I'd rather have a global `sub_state_dict` utility function in `base_lightning.py`.